src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1089,9 +1089,14 @@ def create_or_replace_view(
         source_plan: Optional[LogicalPlan],
     ) -> SnowflakePlan:
         if len(child.queries) != 1:
-            raise SnowparkClientExceptionMessages.PLAN_CREATE_VIEW_FROM_DDL_DML_OPERATIONS()
+            # If creating a temp view, we can't drop any temp object in the post_actions
+            # It's okay to leave these temp objects in the current session for now.
+            if is_temp:
+                child.post_actions.clear()
```

Comment:
Better only remove the post action that drops the temp table while keep other ones.

Reply:
yes good point, added